src/language-html/utils.js

```diff
@@ -232,6 +232,13 @@ function isTrailingSpaceSensitiveNode(node) {
     return true;
   }
 
+  if (
+    node.type === "interpolation" &&
+    node.parent.children.some(child => child.type === "comment")
```

Comment:
This change doesn't seem to have been thought out well. Now whitespace is inserted where it shouldn't be just because those nodes have a sibling comment.

**Prettier pr-7209**
[Playground link](https://deploy-preview-7209--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmo5wDmCAJieRWcMGQCQwAUA5FnjABrPHV4BKMgF8pqAM4CohAOoALAnABucAE5kAnhACuZehAA0p6LxhkAllDm7bAd1V34CwXHvkAZjrQthB+ZBhwAISoAPReSqQsZHIQ4WTwAB4wCSyoEQC0eWSQGOGwZAVMFDHUdFCMpDGYuBqEIOYgEFgwdtByyKB4OoEuAAqDCH0oeJoQdvRtIABGOoJCcDAAygJgDjTIMDpGcO2qMBg4ah5wXmBwGxMedpoe+sjgcn3tDk46MCMrNAweGQfjwOCc7QAVnJ0gAhFbCdYbPDhAAyDjgILBEJA0PSG121AAikYIPAseDjiABDofm9NEcFlgdA4YMo5jBVMgABwABnazIgTmUKywb2Z1102gWAEdSfB-p1JiA8HI8lA4HB6FqFjo4HK7Hr-qIgRScU4MHZ9ocqXJCXASWTMUhQZT2jA8It2fROcgAEzulZ2HC7ADCKVNKCg0Ex7SMTgAKp7Jq6cQy4ABJOoITZgFldACCdQ2MH01DNcBkQA)
```sh
--parser vue
```

**Input:**
```vue
<template>
  <legend>
    {{ $t('package') }}<span>Whatever you do, don't insert whitespace in front of me!</span>
    some text
    <!-- comment -->
  </legend>
</template>
```

**Output:**
```vue
<template>
  <legend>
    {{ $t("package") }}
    <span>Whatever you do, don't insert whitespace in front of me!</span>
    some text
    <!-- comment -->
  </legend>
</template>

```

Reply:
Thanks for reviewing!
I noticed that code is unnecessary, so I removed it.